infrastructure/infrastructure.py

```
@@ -0,0 +1,638 @@
+
+
+def create_docker_context(manager: Instance) -> None:
+    """Create a Docker context using Tailscale DNS name"""
+    tailscale_tailnet_id = os.getenv("TAILSCALE_TAILNET_ID")
+    docker = shutil.which("docker")
+
+    if not docker:
+        logger.error("Docker executable not found")
+        return
+
+    subprocess.run(
```

Ruff (0.13.1)

`subprocess` call: check for execution of untrusted input

(S603)